From 4e1ccdc32959e13b5e86188545f7fe506f121715 Mon Sep 17 00:00:00 2001
From: Pauli Nieminen <pauli.nieminen@linux.intel.com>
Date: Mon, 7 May 2012 21:39:57 +0300
Subject: Default to -fno-omit-frame-pointer

Android assumes it can do frame pointer based unwinding so we have to
disable gcc default optimization. The best possible case would be making
unwinding to use .eh_frames but that is some work because .eh_frame is
subset of dwarf debug information.

To make debugging easier disabling frame pointer optimization makes
sense for now. With this patch I can get working unwinding when crash
happens.

If .eh_frame unwinding is implemented this patch can be reverted to
allow one extra register for compiler optimization.

For: AIA-449
Change-Id: I56e31af1e9c99fb08118923632e44ac195e93c8c
Depends-Change-Id: I544d3cbc4d3cad28441f1b923da5ecd5a6762b33
Signed-off-by: Pauli Nieminen <pauli.nieminen@linux.intel.com>
---
 core/combo/TARGET_linux-x86.mk |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/core/combo/TARGET_linux-x86.mk b/core/combo/TARGET_linux-x86.mk
index 27a1562..7eeacdb 100644
--- a/core/combo/TARGET_linux-x86.mk
+++ b/core/combo/TARGET_linux-x86.mk
@@ -100,6 +100,7 @@ TARGET_GLOBAL_CFLAGS += \
 			-fstrict-aliasing \
 			-funswitch-loops \
 			-funwind-tables \
+			-fno-omit-frame-pointer \
 			-fstack-protector
 
 android_config_h := $(call select-android-config-h,target_linux-x86)
-- 
1.7.4.1

